src/perfetto_cmd/perfetto_cmd.cc

```diff
@@ -768,6 +792,38 @@ std::optional<int> PerfettoCmd::ParseCmdlineAndMaybeDaemonize(int argc,
     }
   }
 
+  if (upload_flag_ && trace_config_->persist_trace_after_reboot()) {
+    if (!trace_config_->write_into_file()) {
+      PERFETTO_ELOG(
+          "TraceConfig's 'write_into_file' must be true when using "
+          "'persist_trace_after_reboot'.");
+      return 1;
+    }
+    if (trace_config_->unique_session_name().empty()) {
+      PERFETTO_ELOG(
+          "TraceConfig's 'unique_session_name' must be set when using "
+          "'persist_trace_after_reboot'.");
+      return 1;
+    }
+    // PerfettoCmd assumes 'trace_out_path_' is the active output file owned by
+    // this process. When perfetto_cmd stops it automatically removes the file
+    // if '!trace_out_path_.empty()' and 'bytes_written_ == 0'.
+    // We use a separate field for the persistent trace file to prevent it from
+    // being automatically removed when perfetto_cmd stops.
```

Comment:
But in the case of a graceful stop, don't you want it to be auotmatically removed?
I'm not sure I understand why we need this special case. can you expand?